src/nrniv/netpar.cpp

```diff
@@ -314,7 +314,7 @@ void NetParEvent::savestate_restore(double tt, NetCvode* nc) {
     if (ithread_ == 0) {
         // npe_->pr("savestate_restore", tt, nc);
         for (int i = 0; i < nrn_nthread; ++i)
-            if (npe_ + i) {
+            if (i < n_npe_) {
```

Comment:
The intent might have been to check if `npe_` is a `nullptr`.

Reply:
You may be correct. But I think the ```if (i < n_npe_)``` is always valid (at least locally at this code position) even if ```npe_``` size does not match with nrn_nthread.